odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1671 @@
+            raise TypeError(
+                'vector {} not in the domain of the functional {}.'
+                ''.format(translation, convex_conj_f.domain))
+
+        super().__init__(domain=convex_conj_f.domain,
+                         linear=convex_conj_f.is_linear)
+
+        # Only compute the grad_lipschitz if it is not inf
+        if not convex_conj_f.grad_lipschitz == np.inf:
+            self._grad_lipschitz = (convex_conj_f.grad_lipschitz +
+                                    translation.norm())
+
+        self._orig_convex_conj_f = convex_conj_f
+        self._translation = translation
+
+    @property
+    def orig_convex_conj_f(self):
+        return self._orig_convex_conj_f
+
+    @property
+    def translation(self):
+        return self._translation
+
+    def _call(self, x):
+        """Applies the functional to the given point.
+
+        Parameters
+        ----------
+        x : `LinearSpaceVector`
+            Element in the domain of the functional.
+            The point in which the functional is evaluated
+
+        Returns
+        -------
+        `self(x)` : `element` in the `field` of the ``domain``.
+            Evaluation of the functional.
+        """
+        return self.orig_convex_conj_f(x) + x.inner(self.translation)
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional.
+
+        Notes
+        -----
+        The operator that corresponds to the mapping
+
+        .. math::
+
+            x \\to \\nabla f(x)
+
+        where :math:`\\nabla f(x)` is the element used to evaluate
+        derivatives in a direction :math:`d` by
+        :math:`\\langle \\nabla f(x), d \\rangle`.
+        """
+        return (self.orig_convex_conj_f.gradient +
+                ConstantOperator(self.translation))
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the ConvexConjugateTranslation
+        functional.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator
+
+        Returns
+        -------
+        out : Operator
+            Domain and range equal to domain of functional
+        """
+        return proximal_quadratic_perturbation(
+            self.orig_convex_conj_f.proximal, a=0, u=self.translation)(sigma)
+
+    @property
+    def conjugate_functional(self):
+        """Convex conjugate functional of the functional.
+
+        Notes
+        -----
+        By the Fenchel-Moreau theorem, the convex conjugate of a convex
+        conjugate is the functional itself, if and only if the functional is
+        proper, convex, and lower semi-continuous.
+
+        Wikipedia article on the `Fenchel-Moreau theorem
+        <https://en.wikipedia.org/wiki/Fenchel%E2%80%93Moreau_theorem>`_.
+        """
+
+        return self.orig_convex_conj_f.conjugate_functional.traslate(
+            self.translation)
+
+
+class ConvexConjugateFuncScaling(Functional):
```

Comment:
At what point is this useful where `scaling * F * (1 / scaling)` is not applicable?